api/mfx_dispatch/linux/mfxparser.cpp

```diff
@@ -131,43 +137,54 @@ void parse(const char* file_name, std::list<PluginInfo>& plugins)
 #else
   const char* mode = "r";
 #endif
-  char line[PATH_MAX];
-  PluginInfo plg;
 
   FILE* file = fopen(file_name, mode);
   if (!file)
     return;
 
-  while(fgets(line, PATH_MAX, file)) {
-    char* p = skip(line);
+  char c_line[PATH_MAX];
+  PluginInfo plg;
+  std::string line;
+
+  while(fgets(c_line, PATH_MAX, file))
+  {
+    line = c_line;
+
+    skip(line);
+    strip(line);
 
-    if (strchr(";#", *p)) {
+    if (line.find_first_of(";#") != std::string::npos)
+    {
       // skip comments
-    } else if (*p == '[') {
+    }
+    else if (line[0] == '[')
+    {
       if (plg.isValid()) {
         plugins.push_back(std::move(plg));
         plg = PluginInfo{};
       }
-    } else {
-      char* name = p;
-      char* value = p;
-
-      while(*value && !strchr("=:;#", *value)) ++value;
-      if (*value && strchr("=:", *value)) {
-        *value++ = '\0';
-      }
-
-      p = value;
-      while (*p && !strchr(";#", *p)) ++p;
-      if (*p != '\0') {
-        *p = '\0';
+    }
+    else
+    {
+      std::string name = line, value = line;
+
+      size_t pos = value.find_first_of("=:");
+      if (pos != std::string::npos)
+      {
+          // Get left part relative to delimiter
+          name.erase(pos);
+          strip(name);
+
+          // Get right part relative to delimiter
+          value.erase(0, pos + 1);
+          skip(value);
+
+          pos = value.find_last_not_of(";#");
```

Comment:
Probably makes sense to introduce strip(line, pattern)

Reply:
I think right now it will be an overkill